src/Reflection/ReflectionClass.php

```diff
@@ -244,6 +244,7 @@ private function createMethodsFromTrait(ReflectionMethod $method): array
         $methodHash = $this->methodHash($method->getImplementingClass()->getName(), $method->getName());
 
         if (array_key_exists($methodHash, $traitModifiers)) {
+            $methodAst        = clone $methodAst;
```

Comment:
Oooooof, that's an ugly one :| 

I suppose AST should be marked `/** @psalm-immutable */` when possible, but too late for that...